pandas/_libs/lib.pyx

```diff
@@ -886,6 +886,22 @@ def is_period(val: object) -> bool:
     return util.is_period_object(val)
 
 
+cpdef inline bint is_list_like(object obj):
```

Comment:
so I can use this impl here: https://github.com/pandas-dev/pandas/blob/master/pandas/core/dtypes/inference.py#L244 

but is ``hasattr(obj, '__iter__')`` equiv to isinstance(abc.Iterable)? can I use the abcs in cython?]

cc @jbrockmendel

Reply:
answer is to share the same code :->